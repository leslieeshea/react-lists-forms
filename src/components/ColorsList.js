import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

function Colors({ colors }) {
  const colorList = colors.map(({ name, hex, red, green, blue }) => (
    <li key={`${name}-${hex}`}>
      <Color  name={name} hex={hex} red={red} green={green} blue={blue} />
    </li>
  ));

  return (
    <ul>
      {colorList}
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};

export default Colors;
