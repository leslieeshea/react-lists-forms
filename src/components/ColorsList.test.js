import React from 'react';
import { shallow } from 'enzyme';
import ColorsList from './ColorsList';

describe('ColorsList component', () => {
  it('renders a list of colors', () => {
    const colors = [
      { name: 'Blue', hex: '#0000FF', red: 0, green: 0, blue: 255 },
      { name: 'Blue', hex: '#FF0000', red: 255, green: 0, blue: 0 }
    ];

    const wrapper = shallow(<ColorsList colors={colors} />);
    expect(wrapper).toMatchSnapshot();
  });
});
