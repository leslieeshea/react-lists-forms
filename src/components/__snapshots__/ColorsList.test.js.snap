// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorsList component renders a list of colors 1`] = `
<ul>
  <li
    key="Blue-#0000FF"
  >
    <Color
      blue={255}
      green={0}
      hex="#0000FF"
      name="Blue"
      red={0}
    />
  </li>
  <li
    key="Blue-#FF0000"
  >
    <Color
      blue={0}
      green={0}
      hex="#FF0000"
      name="Blue"
      red={255}
    />
  </li>
</ul>
`;
